packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -22,109 +31,142 @@ export interface Props {
   noHeader?: boolean;
   resizable?: boolean;
   onCellClick?: TableFilterActionCallback;
-  onColumnResize?: ColumnResizeActionCallback;
+  onColumnResize?: TableColumnResizeActionCallback;
+  onSortBy?: TableSortByActionCallback;
 }
 
-export const Table: FC<Props> = memo(
-  ({ data, height, onCellClick, width, columnMinWidth = COLUMN_MIN_WIDTH, noHeader, resizable = false }) => {
-    const theme = useTheme();
-    const [ref, headerRowMeasurements] = useMeasure();
-    const tableStyles = getTableStyles(theme);
-    const memoizedColumns = useMemo(() => getColumns(data, width, columnMinWidth), [data, width, columnMinWidth]);
-    const memoizedData = useMemo(() => getTableRows(data), [data]);
-
-    const defaultColumn = React.useMemo(
-      () => ({
-        minWidth: memoizedColumns.reduce((minWidth, column) => {
-          if (column.width) {
-            const width = typeof column.width === 'string' ? parseInt(column.width, 10) : column.width;
-            return Math.min(minWidth, width);
+interface ReactTableInternalState extends UseResizeColumnsState<{}>, UseSortByState<{}> {}
+
+function useTableStateReducer(props: Props) {
+  return useCallback(
+    (newState: ReactTableInternalState, action: any) => {
+      console.log(action, newState);
```

Comment:
nit: maybe remove console.log

Reply:
going to remove this in later PR when sort by persistence is done